### PR TITLE
WAF: Handle share data option to write to log

### PR DESCRIPTION
**Pull request by miguelxpn**
#### Changes proposed in this Pull Request:

* With this patch we'll only write to the block log if the share data toggle is enabled.

#### Jetpack product discussion

p1651144025906559-slack-C029WFNV69M

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Enable the waf module.
* Try a request that the waf will block with the share data option turned off.
* Verify that nothing was written to the blocklog file.
* Turn the share data option on.
* Try the same request and verify that the blocklog file was created and the block was logged correctly.
* Bonus points: Verify that the bootstrap code is exporting the contents of the option correctly.

### CODE REVIEW

File: projects/packages/waf/src/class-waf-runner.php [hunk 1]
```diff
@@ -39,6 +40,18 @@ public static function define_mode() {
...
 	}
 
+	/**
+	 * Set the mode definition if it has not been set.
+	 *
+	 * @return void
+	 */
+	public static function define_share_data() {
+		if ( ! defined( 'JETPACK_WAF_SHARE_DATA' ) ) {
+			$share_data_option = get_option( self::SHARE_DATA_OPTION_NAME, false );
```
Comment (singerb) [inline projects/packages/waf/src/class-waf-runner.php:50]: Can we update these defaults to true based on 0-as-1202204517249591/1202211719604597/f ?

Comment (miguelxpn) [reply projects/packages/waf/src/class-waf-runner.php:50]: Yes, but I rather do it in the hooks that initialize the module, if we just change the default to true here we'll have the weird behavior of the option not existing at all and being toggled off in the UI and then the logs being written. I was thinking of just setting the option as true when the module is enabled so it shows as enabled in the UI as well.

Comment (singerb) [reply projects/packages/waf/src/class-waf-runner.php:50]: K.

**Review by singerb — APPROVED** [review]
Looks good on this side, pending fixing up the versions and updating the UI.

**Review by singerb — APPROVED** [review]
LGTM; updating the default to true will be separate?

**Comment by miguelxpn** [comment]
> LGTM; updating the default to true will be separate?

I was going to wait until the module hooks was approved and add it as a follow up, but since we call the activation method of Waf_Runner there we can just add the option as part of it in this PR. I used add_option as that won't update already existing options, so if it's already there it won't be overwritten.

**Review by singerb — DISMISSED** [review]
Yep, that's a good place for it.

**Review by singerb — APPROVED** [review]
Approving again.

**Review by singerb — APPROVED** [review]
Re-approving

**Review by kraftbj — APPROVED** [review]

**Pull request merged**